specification/guestconfiguration/resource-manager/Microsoft.GuestConfiguration/preview/2018-06-30-preview/guestconfiguration.json

```diff
@@ -852,6 +1013,323 @@
+        "properties": {
+          "value": {
+            "type": "array",
+            "items": {
+              "$ref": "#/definitions/SoftwareUpdate"
+            },
+            "description": "The collection of software updates found on the computer."
+          }
+        },
+        "description": "The collection of software updates found on the computer."
+      },
+      "SoftwareUpdate": {
+        "properties": {
+          "properties": {
+            "x-ms-client-flatten": true,
+            "$ref": "#/definitions/SoftwareUpdateProperties",
+            "description": "Properties of the software update."
+          }
+        },
+        "allOf": [
+          {
+          "$ref": "../../common/v1/definitions.json#/definitions/ProxyResource"
+        }
+      ],
+      "description": "Describes a specific software update on the given virtual machine."
+      },
+      "SoftwareUpdateProperties": {
+        "description": "Properties of the software update object.",
+        "properties": {
+          "updateState": {
+            "type": "string",
+            "readOnly": true,
+            "description": "Needed / Installed"
+          },
+          "availableOnComputerSince": {
+            "type": "string",
+            "readOnly": true,
+            "description": "Point in time when this update was first seen on this computer"
+          },
+          "approved": {
+            "type": "string",
+            "readOnly": true,
+            "description": "Approval status"
+          },
+          "packageRepository": {
+            "type": "string",
+            "readOnly": true,
+            "description": "Package repository"
+          },
+          "supercededUpdateIds": {
+            "type": "string",
+            "readOnly": true,
+            "description": "Superseded Update Ids"
+          },
+          "isSupercededUpdate": {
+            "type": "string",
+            "readOnly": true,
+            "description": "Is a superceded update"
+          },
+          "deploymentAction": {
+            "type": "string",
+            "readOnly": true,
+            "description": "Deployment action"
+          },
+          "kbid": {
+            "type": "string",
+            "readOnly": true,
+            "description": "Knowledge Base ID of a Windows Update"
+          },
+          "kbRevisionNumber": {
+            "type": "string",
+            "readOnly": true,
+            "description": "KB revision number"
+          },
+          "publishedDate": {
+            "type": "string",
+            "readOnly": true,
+            "description": "Reported published date for this update"
+          },
+          "productName": {
+            "type": "string",
+            "readOnly": true,
+            "description": "Product name"
+          },
+          "productVersion": {
+            "type": "string",
+            "readOnly": true,
+            "description": "Product version"
+          },
+          "productArch": {
```

Comment:
string enum?

Reply:
The list is not constrained and may change depending on OS, so keeping it free form string.